src/v2/api/services/UserService.ts

```diff
@@ -157,6 +160,22 @@ export class UserService {
         groupId,
       },
     });
+
+    const studentSelective = await this.getSelectiveDisciplines(studentId);
+    const newGroupSelective = await this.groupService.getSelectiveDisciplines(groupId);
+
+    await this.deleteStudentSelective(studentId);
+
+    for (const discipline of studentSelective) {
+      const exist = newGroupSelective.find((d) => d.subjectId === discipline.subjectId);
```

Comment:
In this way you create selective disciplines in new group even if the user did not select them. 
You should create them only under the following condition: **they were chosen by the user, they are not in the new group**